### DISCUSSION TITLE
Single phase flow input template

### DISCUSSION THREAD

**Discussion by asm-fakhrul** [idaholab/moose#25262] [Q&A Modules: General]
The problem in question can be found in the following directory on MOOSE website: modules > thermal_hydraulics > tutorials > single_phase_flow

url: https://mooseframework.inl.gov/modules/thermal_hydraulics/tutorials/single_phase_flow/problem_description.html#introduction

Why is the water coolant on the secondary side not cooling down the helium coolant in the primary loop? Although the hx/wall seems physically not connected to hx/pri and hx/sec when viewed on Paraview, the hx/pri and hx/sec are thermally connected through the hx/wall according to the input file.

I have played around with the mass flow rate in both loops but could not get the primary coolant to cool down. The hx_sec_T_in and hx_sex_T_out are always in thermal equilibrium with the primary loop at the end of the simulation.

Thank you!

**Comment by lindsayad** [comment]
@joshuahansel

**Comment by GiudGiud** [comment]
> Why is the water coolant on the secondary side not cooling down the helium coolant in the primary loop? Although the hx/wall seems physically not connected to hx/pri and hx/sec when viewed on Paraview, the hx/pri and hx/sec are thermally connected through the hx/wall according to the input file.

In THM you can have components connect even though they are not close in the Mesh.
This can be done for various reasons, to make outputs cleaner and more spaced out in Paraview for example, or simply to connect models that were developed independently at different locations in space. 

This simulation marches to steady state. So we do expect to be at thermal equilibrium at the end.
But what you're seeing does not seem right to me. We'll take a look at those inputs

**Comment by joshuahansel** [comment]
Why do you say that the primary coolant isn't cooling down? I ran step 5 of the tutorial, and I see it cool down:

<img width="652" alt="Screenshot 2023-08-22 at 11 18 41 AM" src="https://github.com/idaholab/moose/assets/23108676/6f85b7a1-d204-4e92-ac03-a8a73e29094f">

What quantity are you looking at? I plotted "T".

**Reply by joshuahansel** [reply]
Color range is roughly 610 K to 800 K.

**Comment by asm-fakhrul** [comment]
@joshuahansel, please see the attached images.
I am referring to step 6 and 7 of the tutorial that include the HX. Please see the attached results. In the second case, I have doubled the power, the length of the HX and the mass flow rate. The core is getting hotter and hotter with the HX inlet and outlet temp being stuck at around 400.

Case 1: step 6 template
![100 W 1 m HX](https://github.com/idaholab/moose/assets/142850874/a8680710-3aa8-426d-bb9e-5ae0d0e08c7a)

++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++

Case 2: step 6 template input values changed

![200 W 2 m HX 2x m_dot](https://github.com/idaholab/moose/assets/142850874/6cc8c2bd-f5ec-43f5-8c3f-08eac76c645d)

**Reply by GiudGiud** [reply]
seems to be the opposite here.
Figure 1 shows HX secondary temperatures in the 4000s and Figure 2 in the 400 range.

Anyhow could you please paste text inside triple backquotes next time? Screenshots dont render as well. thx!

**Reply by joshuahansel** [reply]
Could you provide an input file for your second case?

It's interesting that your mass flow rate is in the opposite direction (and several orders of magnitude higher). I wonder if the PID controller is to blame. Still, the small heat transfer is interesting.

**Comment by asm-fakhrul** [comment]
That's correct. In the first case, everything is in thermal equilibrium at around 4000 K i.e. primary outlet temperature and secondary inlet and outlet temperature. In the second case, the core is getting hotter (16,000 K) but the HX (400 K) is not removing the heat.

**Comment by joshuahansel** [comment]
On a general note, I'm running these tutorial steps, and I'm surprised on how poorly they're running. It's clear that they need some work (but I'm unable to spare all of that time right now). If it's helpful to you, there are these Brayton cycle examples as well: https://mooseframework.inl.gov/modules/thermal_hydraulics/modeling_guide/index.html. These I have been personally developed, so I'm more familiar. It's interesting that these tutorial steps are using so few elements in each channel - that's not the way we usually run these simulations. It's also interesting that we're not using a steady-state check (and apparently not reaching a steady state anyway). I would expect the system to physically reach a steady state much sooner in physical time. It may be some kind of control logic loop to blame.

**Comment by asm-fakhrul** [comment]
Okay. Thank you for looking into it for me.

Sorry, seems like uploading *.i file type is not allowed. Let me share the content of the file in most cumbersome way for your reference.

**Reply by joshuahansel** [reply]
Thanks, but I'd still like to get your file directly because I want to diff the file, and your copy-paste changed space stuff. Just rename the extension of the file to .txt instead of .i and then you can upload.

**Comment by asm-fakhrul** [comment]
The input file is attached.

[single_phase_flow_ex06.txt](https://github.com/idaholab/moose/files/12421766/single_phase_flow_ex06.txt)

**Reply by joshuahansel** [reply]
I ran this input file and do not get the results you did:
```
+----------------+----------------+--------------------------------------+----------------+----------------+----------------+----------------+
| time           | core_T_out     | core_heating:power_shape_fn:integral | hx_pri_T_out   | hx_sec_T_in    | hx_sec_T_out   | m_dot_pump     |
+----------------+----------------+--------------------------------------+----------------+----------------+----------------+----------------+
:                :                :                                      :                :                :                :                :
|   1.414094e+05 |   7.275276e+03 |                         3.141593e-04 |   7.327430e+03 |   7.340079e+03 |   7.340125e+03 |   4.426889e-05 |
+----------------+----------------+--------------------------------------+----------------+----------------+----------------+----------------+
```
(and then it failed after this step). Did you change something since you posted your results? If so, please do not do that - else we have a moving target.

I diffed the files, and I see that you've changed closures as well. I recommend that you change one thing at a time and then you can let us know which change causes the issue.

**Reply by asm-fakhrul** [reply]
@joshuahansel, unfortunately, could not recover or recreate the exact input file (after my pc crashed before I could save the file) that gave the results I shared earlier. However, I hope I was able to communicate the issue/concern in the meantime.

**Reply by joshuahansel** [reply]
That's ok. You just need some kind of input file that gives the kind of issue you were seeing before, and then we can start over at that point. Then I would ask that you work your way from the original tutorial file, one change at a time, to see what causes this issue.

**Reply by asm-fakhrul** [reply]
@joshuahansel, I have been trying to reproduce the result I shared earlier (primary coolant outlet temp 16000K, HX temp 400 K) but could not reproduce it. Do not remember the combination of parameters that I put together intentionally to show the extreme temp difference between the primary and the secondary side. But how would you explain the results you are getting? hx_pri and hx_sec inlet and outlet temperatures are all at 734 K, when the global T_in is set at 300 K. How come the hx_sec_T_in is not 300 K? PS: I am new to MOOSE so pardon me if I am missing something.

**Reply by joshuahansel** [reply]
The problem is that the setup is bad. It was originally designed for the smaller power. The system just keeps heating up indefinitely because it's unable to remove the increased heat. The provided temperature value of 300 K is not necessarily reflected at the inlet because the provided value is what is _just outside_ of the inlet. If your flow is slow enough and the heat source is large enough, then the first element's solution can still be far away from that provided value. Ultimately I think what needs to happen is that the secondary side mass flow rate needs to be increased. I don't have a feel for how much. Perhaps you can apply a PID controller on the secondary side mass flow rate to achieve a certain temperature in the primary loop.

**Reply by asm-fakhrul** [reply]
Playing around with the input parameters in the attached file, the core is generating a lot less power (20 W) now and should be cooled down with less mass flow. Would you be able to take a quick look to see if you can cool down the coolant to, say, 500 K, before it re-enters the primary loop? The coolant leaves the core at around 1050 K now. Thank you for your time and help!

[single_phase_flow_ex05.txt](https://github.com/idaholab/moose/files/12528150/single_phase_flow_ex05.txt)

**Reply by joshuahansel** [reply]
I'm sorry for the delay in response - have you figured things out? Otherwise we can have someone look at it. Note that the THM tutorial is currently being modified by @GiudGiud .

**Comment by asm-fakhrul** [comment]
Apologies about the confusion. My laptop crashed/shut down in the middle of the correspondence. So I am not sure if the input file I shared is the exact version that corresponds to the results. When opened the file later, I did found that the input file had a mass flow rate of e-4. Meaning, the file was in the middle of making changes when the laptop crashed. Let me rerun the case and share the results and the input file again to be precisely sure that we all are on the same page.
